src/shared_modules/utils/windowsHelper.h

```diff
@@ -0,0 +1,389 @@
+/*
+ * Wazuh shared modules utils
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * October 24, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifdef WIN32
+
+#ifndef _NETWORK_WINDOWS_HELPER_H
+#define _NETWORK_WINDOWS_HELPER_H
+
+#include <map>
+#include <memory>
+#include <vector>
+#include <system_error>
+#include <winsock2.h>
+#include <windows.h>
+#include <ws2tcpip.h>
+#include <iphlpapi.h>
+#include <versionhelpers.h>
+#include "mem_op.h"
+#include "stringHelper.h"
+	
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+#pragma GCC diagnostic ignored "-Wreturn-type"
+#pragma GCC diagnostic ignored "-Wcast-function-type"
+
+constexpr auto WORKING_ADAPTERS_INFO_BUFFER_SIZE {15000};
+constexpr auto MAX_ADAPTERS_INFO_TRIES {3};
+
+namespace Utils
+{
+    struct IPAddressSmartDeleter
+    {
+        void operator()(IP_ADAPTER_INFO* address)
+        {
+            win_free(address);
+            address = nullptr;
+        }  
+        void operator()(IP_ADAPTER_ADDRESSES* address)
+        {
+            win_free(address);
+            address = nullptr;
+        }
+    };
+
+    typedef UINT (WINAPI *GetSystemFirmwareTable_t)(DWORD, DWORD, PVOID, DWORD);
+    static GetSystemFirmwareTable_t getSystemFirmwareTableFunctionAddress()
+    {
+        GetSystemFirmwareTable_t ret{nullptr};
+        auto hKernel32{LoadLibrary("kernel32.dll")};
+        if (hKernel32)
+        {
+            ret = reinterpret_cast<GetSystemFirmwareTable_t>(GetProcAddress(hKernel32, "GetSystemFirmwareTable"));
+            FreeLibrary(hKernel32);
+        }
+        return ret;
+    }
+
+    typedef NETIOAPI_API (WINAPI *ConvertLengthToIpv4Mask_t)(ULONG, PULONG);
+    static ConvertLengthToIpv4Mask_t getConvertLengthToIpv4MaskFunctionAddress()
+    {
+        ConvertLengthToIpv4Mask_t ret{nullptr};
+        auto hIphlpapi{LoadLibrary("Iphlpapi.dll")};
+        if (hIphlpapi)
+        {
+            ret = reinterpret_cast<ConvertLengthToIpv4Mask_t>(GetProcAddress(hIphlpapi, "ConvertLengthToIpv4Mask"));
+            FreeLibrary(hIphlpapi);
+        }
+        return ret;
+    }
+
+    typedef NETIOAPI_API (WINAPI *GetIfEntry2_t)(PMIB_IF_ROW2);
+    static GetIfEntry2_t getIfEntry2FunctionAddress()
+    {
+        GetIfEntry2_t ret{nullptr};
+        auto hIphlpapi{LoadLibrary("Iphlpapi.dll")};
```

Comment:
this dll exist in the binary import list ?

Reply:
Changing this by getmodulehandle